documentation/vespa-versions.html

```diff
@@ -32,4 +32,20 @@ <h2 id="stored-data">Stored Data</h2>
 For self-hosted systems, it may be necessary to upgrade through each
 minor release rather than in larger leaps to ensure Vespa can read existing data.
 This is a good practice in any case.
+</p><p>
+When changes to the format of the stored data in Vespa are made,
+one cannot downgrade / roll back an upgrade without going through the below list
+to understand steps required:
```

Comment:
I don't see why it is helpful to ask people to go through an empty list. I suggest we spell out our contract instead: We do not guarantee that it is possible to downgrade to an earlier version of Vespa (although in most cases this is fine).

Reply:
Agree, the list was there as an example of what I think we will add in a few weeks when the bm25 feature is released - you agree we will have such a list then like "Vespa-7.x.z: change to index format blabla - make sure you do this when upgrading etc + link" (and we keep that list here)?

Reply:
Do what when upgrading do you mean? What won't work is *downgrading*, nothing special is needed to *upgrade*. I think we should just state what I write above and that's that.

Reply:
earlier, when we made changes to format, we specified that "make sure you upgrade to vespa-x.y.z before upgrading to something higher. also, make sure you cannot downgrade from x.y.z+W to x.y.z-1". Something like that (depending on the change)

I agree you can always upgrade within a minor - but you want to know that going past a specific release, downgrade is guaranteed to fail because of the change - that should be in the list

maybe this is a bit premature - I will leave this for @geirst to ponder on until we are actually making the change. we will have to signal somehow that the format is changing within releases (must keep this in mind for protocol changes, too), but not sure if this is the best place or elsewhere